src/main/java/io/vertx/codegen/TypeInfo.java

```diff
@@ -66,13 +67,25 @@ public static TypeInfo create(Type type) {
           Thread.currentThread().setContextClassLoader(loader);
         }
         if (classType.isEnum()) {
-          return new Class.Enum(fqcn, false, Collections.emptyList(), module, false);
+          return new Class.Enum(
+              fqcn,
+              classType.getDeclaredAnnotation(VertxGen.class) != null,
+              Stream.of(classType.getEnumConstants()).map(Object::toString).collect(Collectors.toList()),
+              module,
+              false);
         } else {
           ClassKind kind = Helper.getKind(classType::getAnnotation, fqcn);
+          List<TypeParamInfo.Class> typeParams = new ArrayList<>();
+          int index = 0;
+          for (java.lang.reflect.TypeVariable<? extends java.lang.Class<?>> var : classType.getTypeParameters()) {
+            typeParams.add(new TypeParamInfo.Class(classType.getName(), index++, var.getName(), Collections.<Variance>emptySet()));
+          }
           if (kind == ClassKind.API) {
-            return new Class.Api(fqcn, true, Collections.emptyList(), null, null, null, module, false);
+            java.lang.reflect.TypeVariable<java.lang.Class<ReadStream>> classTypeVariable = ReadStream.class.getTypeParameters()[0];
```

Comment:
there is lot of code to read, so it's not clear what FQN refers to (in clear : I'm lost :-) ) can you give more context ?